yarn-project/aztec.js/src/wallet/account_wallet.ts

```diff
@@ -0,0 +1,70 @@
+import { Fr } from '@aztec/circuits.js';
+import { ABIParameterVisibility, FunctionAbiHeader, FunctionType } from '@aztec/foundation/abi';
+import { AuthWitness, AztecRPC, FunctionCall, TxExecutionRequest } from '@aztec/types';
+
+import { AccountInterface } from '../account/interface.js';
+import { ContractFunctionInteraction } from '../index.js';
+import { BaseWallet } from './base_wallet.js';
+
+/**
+ * A wallet implementation that forwards authentication requests to a provided account.
+ */
+export class AccountWallet extends BaseWallet {
+  constructor(rpc: AztecRPC, protected account: AccountInterface) {
+    super(rpc);
+  }
+
+  createTxExecutionRequest(execs: FunctionCall[]): Promise<TxExecutionRequest> {
+    return this.account.createTxExecutionRequest(execs);
+  }
+
+  async createAuthWitness(message: Fr | Buffer): Promise<AuthWitness> {
+    message = Buffer.isBuffer(message) ? Fr.fromBuffer(message) : message;
+    const witness = await this.account.createAuthWitness(message);
+    await this.rpc.addAuthWitness(witness);
+    return witness;
+  }
+
+  /**
+   * Returns a function interaction to set a message hash as authorised in this account.
+   * Public calls can then consume this authorisation.
```

Comment:
Being a pain here. But, should we not use `authorize` for the American spelling similar to changes made to `serialize` and `deserialize`? Not sure I did that myself actually, but for consistency 😅